src/t8_cmesh_vtk_reader.hxx

```diff
@@ -0,0 +1,72 @@
+/*
+This file is part of t8code.
+t8code is a C library to manage a collection (a forest) of multiple
+connected adaptive space-trees of general element classes in parallel.
+
+Copyright (C) 2015 the developers
+
+t8code is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+t8code is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with t8code; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/** \file t8_cmesh_vtk_reader.hxx
+* Header for the vtk-reader. 
+*/
+
+#ifndef T8_CMESH_VTK_READER
+#define T8_CMESH_VTK_READER
+
+#include <t8_cmesh.h>
+
+#if T8_WITH_VTK
+#include <vtkSmartPointer.h>
+#include <vtkCellData.h>
+#include <vtkDataSet.h>
+#endif
+
+T8_EXTERN_C_BEGIN ();
+
+/**
+ * Construct a cmesh given a filename and a number of files (for parallel reader).
+ * The \a filename should point to file containing an unstructured Grid.
+ * \warning: This is in production and this header will probably change! Update
+ * as the function progresses
+ * 
+ * \param[in] filename      The name of the file 
```

Comment:
it is hard to understand how i should call this if i have multiple files.
Can you please elaborate in your comment, best with an example.

I am asking myself now: If i have files "mesh_0.vtu, mesh_1.vtu, mesh.pvtu"
What ist the filename i provide?
I can think of three possibilities:

1. All process set filename = "mesh.pvtu"
2. process i sets filename = "mesh_i.vtu". In this case, what happens if we have more or less processes than files?
3. We just provide the prefix filename = "mesh".

Reply:
I updated the documentation of both functions. In this version, we only support to read a single file. The development of a parallel reader is in development on a different branch. There, it is possible to just provide the pvtu/pvtp file, and vtk will handle the reading.